docs/source/types.rst

```diff
@@ -0,0 +1,231 @@
+``ForwardDiffNumber`` Types
+===========================
+
+You should familiarize yourself with `dual numbers`_ and `hyper-dual numbers`_ before reading this document.
+
+.. _`dual numbers`: https://en.wikipedia.org/wiki/Dual_number
+.. _`hyper-dual numbers`: https://adl.stanford.edu/hyperdual/Fike_AIAA-2011-886.pdf
+
+``Partials{T,C}`` 
+-----------------
+
+.. code-block:: julia
+    
+    immutable Partials{T,C}
+        data::C
+        Partials{N}(data::NTuple{N,T}) = new(data)
```

Comment:
To enforce `T == eltype(C)`. It also allows gives newcomers to the code an idea of what `C` is actually supposed to be. I could've used outer constructors here, I guess, but then the `T == eltype(C)` restriction wouldn't be inherent to the type definition.